### ISSUE TITLE
Game.IgnoreUserInputAfterTextTimeoutMs causing trouble for single-liners

### ISSUE THREAD

**Issue by ghost** [adventuregamestudio/ags#1048]
This problem has been reported by a user on forums:

_I noticed that when using **Game.IgnoreUserInputAfterTextTimeoutMs** (with a setting like 1000 Ms), it works as intended and prevents the player from accidentally clicking away a new speech audio/text message if they clicked a few moments after the previous speech message disappeared._

_But this also creates an unintended problem if using the player character to narrate background comments to click on things. For example, if you click the Mouth icon on a rock, the player will say the corresponding message. When said message disappears automatically - not by clicking it away (and there's no subsequent message following it), you cannot click the mouth icon on anything else until the 1000 Ms have elapsed. Can the engine be made to disregard **IgnoreUserInputAfterTextTimeoutMs** setting if there's no new speech message currently on-screen?_

_This happens with all icons. And it also affects right-clicking, meaning that when using the Sierra-style icon interface, you can't right-click to cycle through icons until the 1000 Ms have elapsed either._

**Comment by ivan-mogilko** [comment]
So, I had a very long discussion about this with another user some time ago, and eventually we came to conclusion that for a true proper handling we would need several settings, which define:
- how long to ignore input after the moment very first line appears (because user may casually misclick it as well);
- how long to ignore input between two consecutive lines;
- how long to ignore input after last line (where there's no immediately following line), because user may try to skip last line but instead fire some interaction on screen;
- above leads to - what counts as "consecutive line", in other words, how much time should pass between two lines for them to be considered "consecutive".

And so on, and so on.
All the above is probably possible to implement, which would result in 3-4 new properties, and some new engine logic counting time between speech lines or something.
But it's a part of more general problem, how should text display be controlled in the game. And because every developer wants to do it their way, and there's always someone who wants to customize it more or differently, this may lead to overcomplicated system in the engine. Again, possible, with certain devotion, but hacking it in the current already bad code will be an awful thing to do.

Long story short, I decided to instead add API that lets user control presence of the speech lines on screen, and write skipping behavior they prefer themselves. Plus, such property will be useful for other means, so it's not something niche too.
This was has been implemented as #1212.

And I close this ticket, hoping we now have something to suggest to people who get this problem again.
